great_expectations/experimental/checkpoint.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Sequence
+
+from great_expectations.checkpoint import Checkpoint as LegacyCheckpoint
+from great_expectations.data_context.types.base import CheckpointValidationConfig
+
+if TYPE_CHECKING:
+    from great_expectations.checkpoint.configurator import ActionDict
+    from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+    from great_expectations.data_context.data_context.abstract_data_context import (
+        AbstractDataContext,
+    )
+
+
+# TODO: to be renamed once we're good to move this out of experimental
+class ExperimentalCheckpoint:
+    def __init__(
+        self,
+        name: str,
+        data_context: AbstractDataContext,
+        action_list: Sequence[ActionDict],
+        validations: list[dict] | list[CheckpointValidationConfig],
```

Comment:
`action_list` takes a sequence of typed object, whereas `validations` is a union of `dict` and typed object. This seems a bit inconsistent. Is there a reason why we need `validations` to be a `dict`?

Reply:
I'm not sure where we landed with this but the intent was to allow users to pass dict configs just to reduce the number of imports necessary (everything downstream will use the rich type).

I'd prefer if we enforced just these rich config types but its dependent on how we feel about the extra import.

Reply:
I'm also on team rich config, but if we decide on `dict`s and type coercion we should be consistent and do so for other parameters as well.

Reply:
`CheckpointValidationConfig` has a few fields like `expectation_suite_name` and `batch_request` with typing information